### PR TITLE
[Fix #133] Add new `Minitest/UnreachableAssertion` cop

### LINKED ISSUE #133: New cop suggestion: Detect invalid assertions within an `assert_raises` block
We encountered a test which was misleading because the `assert_equal` was never reached:

```ruby
        assert_raises WidgetError do
          widget = Widget.new # raises WidgetError
          assert_equal "bar", widget.foo # never executed
        end
```

I think as a general rule, an `assert_raises` block shouldn't contain any assertions. Thoughts?

cc @mkevinosullivan

### PR DESCRIPTION
**Pull request by koic**
Fixes #133.

This PR adds new `Minitest/UnreachableAssertion` cop.

This cop checks for `assert_raises` has an assertion method at the bottom of block because the assertion will be never reached.

```ruby
# bad
assert_raises FooError do
  obj.occur_error
  assert_equal('foo', obj.bar) # Never reached.
end

# good
assert_raises FooError do
  obj.occur_error
end
assert_equal('foo', obj.bar)
```

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop-minitest/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop-minitest/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -194,6 +194,11 @@ Minitest/TestMethodName:
   Enabled: 'pending'
   VersionAdded: '0.10'
 
+Minitest/UnreachableAssertion:
+  Description: 'This cop checks for `assert_raises` block contains any assertions.'
```
Comment (andyw8) [inline config/default.yml:198]:
```suggestion
  Description: 'This cop checks if an `assert_raises` block contains any assertions.'
```

Comment (andyw8) [reply config/default.yml:198]: (or: ``This cop check for an `assert_raises` block containing any assertions``)

File: config/default.yml [hunk 2]
```diff
@@ -194,6 +194,11 @@ Minitest/TestMethodName:
   Enabled: 'pending'
   VersionAdded: '0.10'
 
+Minitest/UnreachableAssertion:
+  Description: 'This cop check for an `assert_raises` block containing any unreaclable assertions.'
```
Comment (andyw8) [inline config/default.yml:198]:
```suggestion
  Description: 'This cop checks for an `assert_raises` block containing any unreachable assertions.'
```
(sorry, missed the 's' in my prior comment.)

File: test/rubocop/cop/minitest/unreachable_assertion_test.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  def test_does_not_register_offense_when_not_using_assertion_method_in_assert_raises_block
+    assert_no_offenses(<<~RUBY)
+      assert_raises FooError do
+        do_something('foo', obj.bar)
+      end
+    RUBY
+  end
+
+  def test_registers_offense_when_using_empty_block_for_assert_raises
```
Comment (andyw8) [inline test/rubocop/cop/minitest/unreachable_assertion_test.rb:43]: Should this say `does_not_register_offense`?

Comment (koic) [reply test/rubocop/cop/minitest/unreachable_assertion_test.rb]: This is a typo. I fixed it. Thank you!

**Review by andyw8 — APPROVED** [review]
👍

**Pull request merged**